### ISSUE TITLE
Pick doesn't validate correctly

### ISSUE THREAD

**Issue by guohaolay** [microsoft/TypeScript#55735]
### 🔎 Search Terms
Pick, linter

### 🕗 Version & Regression Information

- This is the behavior in version 4.0.7, and whatever the TypeScript playground uses

### ⏯ Playground Link

https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgCoHsAm7kG8BQyyYwYANhAFzIDOYUoA5gNyHKYQ0IMAOJ6IanQYgWbBOgC2PCpEzUARunQU4IVgF98+MAE8eKDNgAKUCADdgEAO7IAvMmPAEAawA8R9ABpkAIhLkEL7IAD5+EtKyEJi+AHys+BIgdOxwYHCUnvZ4bAEU1L4AwqogyFDKkr5e4lIyEHLU8GQ0ENVEHFy8-IK+AIwATADMVfgaCUkpYFjo1J6mFla2DphpcAlT2KwA9FvIAHoA-EA

### 💻 Code

```ts
interface Todo {
  title: string;
  description: string;
  completed: boolean;
}

type TodoPreview = Pick<Todo, "title" | "completed">;

const data:Todo = {
  title: "Clean room",
  completed: false,
  description:"123",
};

// TypeScript correctly identifies the extra key `description`:
const todo: TodoPreview = {
  title: "Clean room",
  completed: false,
  description:"123",
};

// But doesn't pick up if assigned this way:
const todo2: TodoPreview = data;
```

### 🙁 Actual behavior

No error reported

### 🙂 Expected behavior

Error should be reported

### Additional information about the issue

Pick does not validate correctly if using an object with more keys than the Picked key.

**Comment by MartinJohns** [comment]
This is working as intended. Objects are allowed to have additional properties. Only in specific cases (like assignment of object literals) an "excess property check" is performed, but this is more of a linter feature than part of the type system.

**Comment by guohaolay** [comment]
Thanks for the quick response (and I fixed my ticket, sorry about that).

While I agree that objects can have additional properties, this inconsistency is confusing and conflicting.

**Comment by MartinJohns** [comment]
It's not inconsistent, and confusing only for beginners. TypeScript has a structural type system, and only object literals get excess property checks.

You might be interested in #12936, but don't expect it any time soon.

**Comment by guohaolay** [comment]
I thought TypeScript's mission is to help all levels of developers, to say it's confusing only for beginners kind of defeats the purpose of having it. That proposal does looks great, thanks for sharing.

**Comment by fatcerberus** [comment]
@guohaolay It's known that the current error message for excess property check is confusing and suggests a type rule that doesn't actually exist.  The message will change in 5.3, see https://github.com/microsoft/TypeScript/pull/55152
